src/Components/ProductOverview/ImageGallery/ImageGallery.jsx

```diff
@@ -1,37 +1,147 @@
-/* eslint-disable */
+/*eslint-disable*/
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import './ImageGallery.scss';
 
 export class ImageGallery extends Component {
   constructor(props) {
-      super(props);
-      this.state={};
+    super(props);
+    this.state = {
+      selectedIndex: 0,
+      thumbnailIndex: 0,
+    };
+    this.resetSelectedIndex = this.resetSelectedIndex.bind(this);
+    this.thumbnailClick = this.thumbnailClick.bind(this);
+    this.scrollPrevious = this.scrollPrevious.bind(this);
+    this.scrollNext = this.scrollNext.bind(this);
+    this.showNextOrPrevious = this.showNextOrPrevious.bind(this);
+  }
+
+  resetSelectedIndex() {
+    this.setState({ selectedIndex: 0 });
+  }
```

Comment:
Function to avoid bug of having a higher selectedIndex than is available for a different style that is selected

Reply:
this is exactly what i'm still trying to get my carousel to do right now, however I'm trying to do it without local state.  Have you tried it using redux yet?  It should be just a simple counter with reset (for the edge case you describe here) but I can't get mine to work properly